### PR TITLE
Define structs and functionize request parsing.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

**Describe what this PR does**
This PR functionize parsing request and fill data on subjected structs. 

**Is there anything that requires special attention?**
Do you have any questions? Did you do something clever?

**Related issues:**
Mention any github issues relevant to this PR

### CODE REVIEW

**Comment by humblec** [comment]
@Madhu-1 Please review

File: pkg/glusterfs/controllerserver.go [hunk 1]
```diff
@@ -29,6 +29,36 @@ type ControllerServer struct {
 	*GfDriver
 }
 
+// csiDrvParam stores csi driver specific request parameters.
+type csiDrvParam struct {
+	glusterCluster string
```
[the remark this answers is not in the document]
Comment (humblec) [reply pkg/glusterfs/controllerserver.go]: @Madhu-1 this strcut will be used to gather specific to CSI driver (`csiDrvName`, `csiDrvVersion`..etc) and also gather parameters passed from SC and not part of `gd2volcreate` api. That said, `glusterCluster` will be the resturl and `glusteruser` will be the `username` who got access to the APIs. `glusterVersion` is the parameter which says the version of the glustercluster given in resturl endpoint. 
`compMatrix` is going to be a map which will be internally defined by the driver to make a compatibility version matrix between CSI driver and gluster cluster. All these fields are optional and can be used if needed.

File: pkg/glusterfs/controllerserver.go [hunk 2]
```diff
@@ -29,6 +29,44 @@ type ControllerServer struct {
...
+// and also gather parameters passed from SC and not part of gd2volcreate api.
+// glusterCluster - The resturl of gluster cluster
+// glusterUser - The gluster username who got access to the APIs.
+// glusterVersion - Says the version of the glustercluster running in glusterCluster endpoint.
+// compMatrix - map which will be internally defined by the driver to make a compatibility
+//              version matrix between CSI driver and gluster cluster. All these fields are optional and can be used if needed.
+
+type csiDrvParam struct {
+	glusterCluster string
+	glusterUser    string
```
Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: @humblecI as discussed to make more sense we need to add the userToken field to the struct.
as we require both username and Token to connect to rest endpoints

Comment (humblec) [reply pkg/glusterfs/controllerserver.go]: @Madhu-1 awesome, adding that too.

**Comment by humblec** [comment]
@Madhu-1 Addressed review comment. PTAL.

File: pkg/glusterfs/controllerserver.go [hunk 3]
```diff
@@ -56,16 +97,21 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
...
-	volumeName := req.Name
-	glusterVol := req.GetParameters()["glustervol"]
-	glusterServer = req.GetParameters()["glusterserver"]
-	glusterURL := req.GetParameters()["glusterurl"]
-	glusterURLPort := req.GetParameters()["glusterurlport"]
-	glusterUser := req.GetParameters()["glusteruser"]
-	glusterUserSecret := req.GetParameters()["glusterusersecret"]
+	// parse the request.
+	parseResp, parseErr := cs.ParseRequest(req)
+	if parseErr != nil {
```
Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: as we are always returning `nil` do we need error handling?

Comment (humblec) [reply pkg/glusterfs/controllerserver.go]: the parseErr will have errors soon :) @Madhu-1

File: pkg/glusterfs/controllerserver.go [hunk 4]
```diff
@@ -56,16 +97,21 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
...
+		return nil, status.Error(codes.InvalidArgument, "failed to parse request")
+	}
+
+	if parseResp != nil && parseResp.gdVolReq != nil {
```
Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: same here as well, as we are not returning nil here, we don't need this one also

Comment (humblec) [reply pkg/glusterfs/controllerserver.go]: @Madhu-1 same as above.

**Comment by humblec** [comment]
Thanks @Madhu-1 for review!! I am merging this as this is the base for few other PRs.

**Pull request merged**